tests/std/tests/P0896R4_views_take/test.cpp

```diff
@@ -506,6 +508,40 @@ void test_DevCom_1397309() {
 
     assert(ranges::equal(values | ranges::views::take(2) | ranges::views::keys, expected));
 }
+struct read_some_int_range : ranges::subrange<counted_iterator<istream_iterator<int>>, default_sentinel_t> {
+    using ranges::subrange<counted_iterator<istream_iterator<int>>, default_sentinel_t>::subrange;
+};
+
+template <>
+inline constexpr bool ranges::disable_sized_range<read_some_int_range> = true;
+
+void test_lwg3737() {
+    static_assert(ranges::input_range<read_some_int_range>);
+    static_assert(ranges::input_range<const read_some_int_range>);
+    static_assert(!ranges::sized_range<read_some_int_range>);
+    static_assert(!ranges::sized_range<const read_some_int_range>);
+
+    istringstream stream{"0 1 42 1729"};
+    auto rng =
+        read_some_int_range{counted_iterator{istream_iterator<int>{stream}, 4}, default_sentinel} | views::take(2);
+
+    using result_range = decltype(rng);
+    static_assert(
+        is_same_v<ranges::iterator_t<result_range>, counted_iterator<counted_iterator<istream_iterator<int>>>>);
+    static_assert(is_same_v<ranges::sentinel_t<result_range>, default_sentinel_t>);
+
+    static_assert(
+        is_same_v<ranges::iterator_t<const result_range>, counted_iterator<counted_iterator<istream_iterator<int>>>>);
+    static_assert(is_same_v<ranges::sentinel_t<const result_range>, default_sentinel_t>);
+
+    vector<int> vec{};
+    for (const int elem : rng) {
+        vec.push_back(elem);
+    }

```

Comment:
We can't `append_range` since this needs to compile as c++20, but we could:
```suggestion
    ranges::copy(rng, back_inserter(vec));
```
to replace this raw loop.